src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
+  DataSet *d = reinterpret_cast<DataSet *>(dataset);
+
   s->put_dataset(*d);
-  return;
 }
 
+// Return a pointer to a new dataset.  The user is
+// responsible for deleting the dataset via DeallocateeDataSet()
 extern "C"
-void* get_dataset(void* c_client, const char* name,
-                  const size_t name_length)
+void *get_dataset(void *c_client, const char *name, const size_t name_length)
 {
-  /* Return a pointer to a new dataset.  The user is
-  responsible for deleting the dataset.
-  */
-  Client* s = (Client *)c_client;
-  std::string dataset_name = std::string(name, name_length);
-  DataSet* dataset = new DataSet(s->get_dataset(dataset_name));
-  return (void*)dataset;
+  // Sanity check params
+  if (NULL == c_client || NULL == name)
+    return NULL; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string dataset_name(name, name_length);
+  DataSet *dataset = NULL;
+
+  try {
+    dataset = new DataSet(s->get_dataset(dataset_name));
+  } catch (...) {
+    dataset = NULL;
+  }
+  return reinterpret_cast<void *>(dataset);
 }
 
+// Rename a dataset in the database.
 extern "C"
-void rename_dataset(void* c_client, const char* name,
-                    const size_t name_length, const char* new_name,
+void rename_dataset(void *c_client, const char *name,
+                    const size_t name_length, const char *new_name,
                     const size_t new_name_length)
 {
-  /* Rename a dataset in the database.
-  */
-  Client* s = (Client *)c_client;
-  std::string name_str = std::string(name, name_length);
-  std::string new_name_str = std::string(new_name, new_name_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == name || NULL == new_name)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string name_str(name, name_length);
+  std::string new_name_str(new_name, new_name_length);
+
   s->rename_dataset(name_str, new_name_str);
-  return;
 }
 
+// Copy a dataset from the src_name to the dest_name
 extern "C"
-void copy_dataset(void* c_client, const char* src_name,
-                  const size_t src_name_length, const char* dest_name,
-                  const size_t dest_name_length
-                  )
+void copy_dataset(void *c_client, const char *src_name,
+                  const size_t src_name_length, const char *dest_name,
+                  const size_t dest_name_length)
 {
-  /* Copy a dataset from teh src_name to the dest_name
-  */
-  Client* s = (Client *)c_client;
-  std::string src_name_str = std::string(src_name, src_name_length);
-  std::string dest_name_str = std::string(dest_name, dest_name_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == src_name || NULL == dest_name)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
```

Comment:
Same as before.  Comment above to limit long lines

Reply:
Whatever we agree to I'll do universally